### ISSUE TITLE
Annotation disappears on subsequent re-renders

### ISSUE THREAD

**Issue by franky47** [linkstrifer/react-rough-notation#6]
I'll have to take a closer look later, but it appears that when re-rendering the component, the animation state goes from `showing` to `not-showing`, here's an example where the rerender is triggered by a theme change:
![CleanShot 2020-05-30 at 07 23 43](https://user-images.githubusercontent.com/1174092/83320321-a9e52a80-a246-11ea-9fb4-8a3124e87a47.gif)

**Comment by franky47** [comment]
Ok I think I've got it: 
https://github.com/linkstrifer/react-rough-notation/blob/ea5e05887b40882095eda420f998c0374e6d6e7a/src/RoughNotation.tsx#L68

The `configuration` passed in the dependency array of the first `React.useEffect` always changes between renders, which triggers the returned callback, removing the annotation. Because `show` did not change, the second `useEffect` does not run, effectively not showing the newly created annotation.

My first attempt at a fix was to make sure the contents of configuration did not change, rather than the reference itself. For that, using `JSON.stringify(configuration)` as a dependency does the trick, but memoizing the configuration creation would probably be more robust.

However this is the wrong thing to solve. If something indeed changed in the configuration, but `show` did not change, then the newly created (updated) annotation will not be shown, as the second `useEffect` will not run. But now, if we replace `annotation` in the second `useEffect` dependency (which is not needed, it's a Ref which does not change) with the `JSON.stringify(configuration)`, any prop update does redraw the annotation, solving (almost) everything.

Almost, because changing things like the color of the annotation triggers a redraw, which will not look the same as the previous one, as intended by Rough.js. Technically, only `color` and `strokeWidth` should be updatable without require a regeneration of the annotation object, but I'm not yet familiar with rough-notation's API to know if they handled that case.

Edit: here's what I have with an annotation that changes color after 5s, then changing the theme (re-render without any props change):
![CleanShot 2020-05-30 at 16 11 38](https://user-images.githubusercontent.com/1174092/83330612-7b8c3d00-a290-11ea-809c-37797f11fb54.gif)

As I'll want to change the color of the annotation based on the theme, I'd rather it did not regenerate or re-animate. I'll look into it.

**Comment by linkstrifer** [comment]
Wow ... awesome research, I found a similar problem creating the playground, I would like to change the line properties (color, width) without a redraw but right now `rough-annotation` doesn't have a way to change it, at least not documented, I'm going to check the code today to see if is possible.

You are right about the `configuration` issue on every render, maybe put it in a `useState` and check if `color` or `strokeWidth` changed to update the config is a good idea, what do you think?

**Comment by franky47** [comment]
From what I read on Rough.js and `rough-annotation`, there's no way to mutate a generated SVG once it has been generated. It can be targeted with external CSS though, I'm looking into this in my app. 

The alternative is what Excalidraw does (which allows changing object colors and properties), which is to use a Generator to ensure the randomness that defines an output remains constant (more info in [this video](https://www.youtube.com/watch?v=fix2-SynPGE&feature=youtu.be&t=421)).

For the configuration changes, this could work:
```tsx
useEffect(() => {
  annotation.current = annotate(element.current, {
    animate,
    animationDelay,
    animationDuration,
    color,
    padding,
    strokeWidth,
    type,
  })
  
  if (getAnnotationObject) {
    getAnnotationObject(annotation.current)
  }

  return () => {
    annotation.current.remove()
  }
}, [
  animate,
  animationDelay,
  animationDuration,
  color,
  padding,
  strokeWidth,
  type,
  getAnnotationObject,
])
```

**Comment by linkstrifer** [comment]
Fix added in v0.0.7, thanks for the help

**Comment by franky47** [comment]
97005e7 only solved half of the problem I'm afraid, you'll also need to show the newly created annotation when one of the props has changed.

Example:
https://codesandbox.io/s/react-rough-notation-reactivity-zuheb?file=/src/App.js

**Comment by linkstrifer** [comment]
Damn ... so close, reopening

**Comment by franky47** [comment]
In the mean time, I use this workaround:
```tsx
// Wrap in a useCallback to keep a constant ref across renders:
const showOnChange = React.useCallback((annotation: any) => {
  annotation.show()
}, [])

return (
  <RoughNotation
    show
    getAnnotationObject={showOnChange}
    {...props}
  />
)
```

However, it works only for annotations that are always shown.

**Comment by franky47** [comment]
Also, it looks like the annotation is recreated when the viewport size changes (I'm changing the window width off-screen here):
![CleanShot 2020-06-04 at 13 52 08](https://user-images.githubusercontent.com/1174092/83753509-b87a7a00-a66a-11ea-9e02-e2304cecf42b.gif)

Preview: https://francoisbest.com/design-system#annotations

**Comment by linkstrifer** [comment]
I think that is expected from the routh-notation library, to re-calculate positions, is resize a valid user case for you?

**Comment by linkstrifer** [comment]
They added property changes in https://github.com/pshihn/rough-notation/pull/31, I will import it to this library and check if it need fixes

**Comment by linkstrifer** [comment]
Added update styling support in `v0.0.9`

**Comment by bogdankjastrzebski** [comment]
Don't know if this one's related, but I encountered a problem of rough-notation not redrawing after view transition in Astro.

Here's the example: [link](https://astro-theme-mia.pages.dev).
Rough notation underscores etc. appear only the first time a post is viewed.
1. View a post.
2. Go to Posts list.
3. View the post again.

The issue is related to view transitions. If we remove view transitions (remove <ViewTransition /> in the meta.astro) the page reloads each time a link is clicked, and everything works. I also tried redrawing rough-notation elements, but it does not seem to work.

Any ideas how to approach the issue?

Edit: Here's the repo with code: https://github.com/thinkininfinity/astro-theme-mia
